sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -523,7 +530,8 @@ object DataSource {
 
   /** Given a provider name, look up the data source class definition. */
   def lookupDataSource(provider: String): Class[_] = {
-    val provider1 = backwardCompatibilityMap.getOrElse(provider, provider)
+    val provider1 = builtinShortNamesMap.getOrElse(provider,
+      backwardCompatibilityMap.getOrElse(provider, provider))
```

Comment:
Should we maybe combine both `builtinShortNamesMap` and `backwardCompatibilityMap` and use a single `getOrElse`? It seems probably confusing to read a bit.